modules/vpc/main.tf

```
@@ -0,0 +1,199 @@
+    description = "allow external ports for SSH, HTTPS, and Automation Mesh"
+  }
+
+  ingress {
+    from_port   = 80
+    to_port     = 80
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+    description = "allow external ports for SSH, HTTPS, and Automation Mesh"
+  }
+
+  ingress {
+    from_port   = 443
+    to_port     = 443
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+    description = "allow external ports for SSH, HTTPS, and Automation Mesh"
+  }
+
+  ingress {
+    from_port   = 27199
+    to_port     = 27199
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+    description = "allow external ports for SSH, HTTPS, and Automation Mesh"
+  }
+
+  ingress {
+    from_port   = -1
+    to_port     = -1
+    protocol    = "icmp"
+    cidr_blocks = [var.infrastructure_vpc_cidr]
+    description = "allow ping on local net"
+  }
+
+  ingress {
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = [var.infrastructure_vpc_cidr]
+    description = "allow aap ports on local net"
+  }
+
+  ingress {
+    from_port   = 80
+    to_port     = 80
+    protocol    = "tcp"
+    cidr_blocks = [var.infrastructure_vpc_cidr]
+    description = "allow aap ports on local net"
+  }
+  ingress {
+    from_port   = 443
+    to_port     = 443
+    protocol    = "tcp"
+    cidr_blocks = [var.infrastructure_vpc_cidr]
+    description = "allow aap ports on local net"
+  }
+  ingress {
+    from_port   = 5432
+    to_port     = 5432
+    protocol    = "tcp"
+    cidr_blocks = [var.infrastructure_vpc_cidr]
+    description = "allow aap ports on local net"
+  }
+  ingress {
+    from_port   = 8443
+    to_port     = 8443
+    protocol    = "tcp"
+    cidr_blocks = [var.infrastructure_vpc_cidr]
+    description = "allow aap ports on local net"
+  }
+  ingress {
+    from_port   = 27199
+    to_port     = 27199
+    protocol    = "tcp"
+    cidr_blocks = [var.infrastructure_vpc_cidr]
+    description = "allow aap ports on local net"
+  }
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+    description = "allow all outbound"
+  }
+
+  tags = merge( 
+      {
+        Name = "aap-infrastructure-${var.deployment_id}-sg"
+      },
+      var.persistent_tags
+    )
+  depends_on = [aws_vpc.aap_infrastructure_vpc]
+}
+
+#
+# Route table
+#
+resource "aws_route_table" "aap_infrastructure_route_table" {
+  vpc_id = aws_vpc.aap_infrastructure_vpc.id
+
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = aws_internet_gateway.aap_infrastructure_igw.id  
+  }
+
+  tags = merge( 
+      {
+        Name = "aap-infrastructure-${var.deployment_id}-rt"
+      },
+      var.persistent_tags
+    )
+  depends_on = [aws_vpc.aap_infrastructure_vpc]
+}
+
+#
+# Route table association
+#
+resource "aws_route_table_association" "aap_infrastructure_subnet_association" {
+  for_each = { for k, subnet in aws_subnet.aap_infrastructure_subnets : k => subnet.id }
```

```suggestion
  for_each = { for key, subnet in aws_subnet.aap_infrastructure_subnets : key => subnet.id }
```
Assuming k meant key.

---

yes k is key